controllers/volumereplicationgroup_controller.go

```diff
@@ -578,55 +578,32 @@ func (v *VRGInstance) validateVRGMode() error {
 	return nil
 }
 
-func (v *VRGInstance) clusterDataRestore(result *ctrl.Result) error {
-	if v.instance.Spec.PrepareForFinalSync || v.instance.Spec.RunFinalSync {
-		msg := "PV restore skipped, as VRG is orchestrating final sync"
-		setVRGClusterDataReadyCondition(&v.instance.Status.Conditions, v.instance.Generation, msg)
-
-		return nil
-	}
+func (v *VRGInstance) clusterDataRestore(result *ctrl.Result) (int, error) {
+	v.log.Info("Restoring PVs and PVCs")
 
-	clusterDataReady := findCondition(v.instance.Status.Conditions, VRGConditionTypeClusterDataReady)
-	if clusterDataReady != nil {
-		v.log.Info("ClusterDataReady condition",
-			"status", clusterDataReady.Status,
-			"reason", clusterDataReady.Reason,
-			"message", clusterDataReady.Message,
-			"observedGeneration", clusterDataReady.ObservedGeneration,
-			"generation", v.instance.Generation,
-		)
-
-		if clusterDataReady.Status == metav1.ConditionTrue &&
-			clusterDataReady.ObservedGeneration == v.instance.Generation {
-			v.log.Info("VRG's ClusterDataReady condition found. PV restore must have already been applied")
-
-			return nil
-		}
-	} else {
-		v.log.Info("ClusterDataReady condition absent")
-	}
-
-	err := v.restorePVsForVolSync()
+	numRestoredForVS, err := v.restorePVsAndPVCsForVolSync()
 	if err != nil {
-		v.log.Info("VolSync PV restore failed")
+		v.log.Info("VolSync PV/PVC restore failed")
 
-		result.Requeue = true
-
-		return fmt.Errorf("failed to restore PVs for VolSync (%w)", err)
+		return numRestoredForVS, fmt.Errorf("failed to restore PV/PVC for VolSync (%w)", err)
 	}
 
-	err = v.clusterDataRestoreForVolRep(result)
+	numRestoredForVR, err := v.restorePVsAndPVCsForVolRep(result)
 	if err != nil {
-		v.log.Info("VolRep ClusterData restore failed")
+		v.log.Info("VolRep PV/PVC restore failed")
 
-		return fmt.Errorf("failed to restore ClusterData for VolRep (%w)", err)
+		return numRestoredForVR, fmt.Errorf("failed to restore PV/PVC for VolRep (%w)", err)
```

Comment:
Should we return `numRestoredForVS` + `numRestoredForVR`? In case `numRestoredForVR` is 0 then status update will not kick in from the caller.

Although caller will get an error and anyway update status, so should not affect overall end result.

Reply:
Yes, to be correct, we should return `forVR` as well even though the error will mask any return value.